### ISSUE TITLE
Prob function location and offset

### ISSUE THREAD

**Issue by TomTranter** [PMEAL/OpenPNM#498]
What is the purpose of offset in the pore and throat size distributions? As far as I can see it performs exactly the same role as location - at least for weibull - do we need both?

**Comment by jgostick** [comment]
it adds a value to the lowest generated number.  I think location sets the
peak of the distribution, but would still give you 0 numbers.

**Comment by TomTranter** [comment]
`import OpenPNM as op
from OpenPNM.Geometry import models as gm
import numpy as np
pn = op.Network.Cubic(shape=[5,5,5])
geom = op.Geometry.GenericGeometry(network=pn,pores=pn.Ps,throats=pn.Ts)
shape = 2.5
scale = 1.0
geom.models.add(propname='pore.seed',
                model=gm.pore_misc.random,
                regen_mode='constant')
geom.models.add(propname='pore.location',
                model=gm.pore_diameter.sphere,
                psd_name='weibull_min',
                psd_shape=shape,
                psd_loc=0.5,
                psd_scale=scale,
                psd_offset=0.0)
geom.models.add(propname='pore.offset',
                model=gm.pore_diameter.sphere,
                psd_name='weibull_min',
                psd_shape=shape,
                psd_loc=0.0,
                psd_scale=scale,
                psd_offset=0.5)
geom.models.add(propname='pore.both',
                model=gm.pore_diameter.sphere,
                psd_name='weibull_min',
                psd_shape=shape,
                psd_loc=0.25,
                psd_scale=scale,
                psd_offset=0.25)
r = 15
a = np.around(geom["pore.location"],r)
b = np.around(geom["pore.offset"],r)
c = np.around(geom["pore.both"],r)
print(np.allclose(a,b),np.allclose(a,c))`

**Comment by TomTranter** [comment]
[location_offset_test.txt](https://github.com/PMEAL/OpenPNM/files/119358/location_offset_test.txt)

**Comment by jgostick** [comment]
Ok, the pore and throat diameter models always bugged me.  I'm not sure we called them 'sphere' and 'cube', duh...that only matters when you find the volume.  Also, the arguments were not consistent between different scipy.stats functions, so basically only weibull worked.  I have created PR #504 to introduce a whole new approach to pore diameter models.  There are new a few specific models for each type of distribution, plus a generic version which accepts a pre-filled (rv_frozen) distribution from scipy.stats and just calls the ppf method.  This PR is much clearer overall, and we should phase out the sphere and cube models.
